src/Drupal/DrupalExtension/Context/ConfigContext.php

```diff
@@ -107,6 +107,13 @@ public function setConfig($name, $key, $value)
     {
         $backup = $this->getDriver()->configGet($name, $key);
         $this->getDriver()->configSet($name, $key, $value);
-        $this->config[$name][$key] = $backup;
+        if (!array_key_exists($name, $this->config)) {
+            $this->config[$name][$key] = $backup;
+            return;
+        }
+
+        if (!array_key_exists($key, $this->config[$name])) {
```

Comment:
As a very small optimization we can rewrite both if statements as one `isset()`:

```
if (!isset($this->config[$name][$key])) {}
```

Reply:
```
php > $array['key'] = NULL;
php > var_dump(isset($array['key']));
bool(false)
```

;)